app/src/lib/stores/github-user-store.ts

```diff
@@ -126,13 +141,23 @@ export class GitHubUserStore extends BaseStore {
    */
   public async getMentionableUsersMatching(
     repository: GitHubRepository,
-    text: string,
-    maxHits: number = 100
+    query: string,
+    maxHits: number = 5
   ): Promise<ReadonlyArray<IMentionableUser>> {
-    const users = await this.getMentionableUsers(repository)
+    assertPersisted(repository, this.getMentionableUsers.name)
+
+    const cache = this.queryCache
+    let users
+
+    if (cache !== null && cache.repository.dbID === repository.dbID) {
+      users = cache.users
+    } else {
+      users = await this.getMentionableUsers(repository)
+      this.setQueryCache(repository, users)
```

Comment:
What do you think about calling `this.setQueryCache()` even if there's a cache hit?

It's a very small detail since the cost of a cache miss is so low, but I think having the cache expiration based on reads makes more sense in this case (we probably don't want the cache to expire when the user types a couple of usernames, which will take longer than a minute).

Reply:
I think that's a great idea

Reply:
https://github.com/desktop/desktop/pull/10121/commits/42584ae7fc6ba6dd5ab3c46b5d5c834358c3b380